### ISSUE TITLE
Can’t arm/disarm: “Web Portal version 23.0.0-99 detected”

### ISSUE THREAD

**Issue by ibogost** [mrjackyliang/homebridge-adt-pulse#82]
Sensors reporting successfully but unable to arm or disarm via plugin anymore. The message below appears in homebridge logs; not sure what it means or what action to take. 

```sh

Web Portal version 23.0.0-99 detected.
Test plugin to ensure system compatibility

```

**Comment by knedd** [comment]
I am having a similar issue. 

[5/22/2022, 7:16:04 AM] [ADT Pulse] Setting Security Panel (system-1) status from stay to off...
[5/22/2022, 7:16:04 AM] [ADT Pulse] Set device status failed.

**Comment by mrjackyliang** [comment]
Hmm, let me check on that

**Comment by nikunp** [comment]
I'm having the same issue as well, but as soon as I arm disarm from the pulse app it updates the status in HomeKit

**Comment by mrjackyliang** [comment]
I'm pretty sure it's a glitch like the past.

**Comment by kmnedd** [comment]
A glitch? In the API?

**Comment by knedd** [comment]
A glitch? In the API?

**Comment by nikunp** [comment]
@mrjackyliang any idea of a fix?

**Comment by mrjackyliang** [comment]
> A glitch? In the API?
> 
> 
> 
> The ADT Pulse app is able to arm and disarm without issue.  So we know the function works.

Yes, because it actually uses the ADT Pulse API. Their API isn't open to the public, so it relies on pretty much hacking the web portal.

**Comment by mrjackyliang** [comment]
> @mrjackyliang any idea of a fix?

Let me check again tomorrow

**Comment by Fergie2020** [comment]
Can confirm, I am having the same issue.

**Comment by jastorino82** [comment]
I have the same issue - HomeKit seems to report the status of the ADT panel, but is unable to arm/disarm. In the logs I get 

ADT Pulse: Portal sync success.
[23/05/2022, 12:12:04] [ADT Pulse] Setting Security Panel (system-1) status from disarmed to stay...
ADT Pulse: Performing portal sync...
ADT Pulse: Setting device status...
ADT Pulse: Response path -> /myhome/23.0.0-99/Ajax/SyncCheckServ?t=1653322324055
ADT Pulse: Response path matches -> true
ADT Pulse: Portal sync success.
ADT Pulse: Response path -> /myhome/23.0.0-99/access/error.jsp
ADT Pulse: Response path matches -> false
ADT Pulse: Set device status to stay failed.
[23/05/2022, 12:12:05] [ADT Pulse] Set device status failed.

**Comment by nikunp** [comment]
@mrjackyliang any idea of a workaround outside of using the adt pulse app while a solution is found?

**Comment by mrjackyliang** [comment]
> I have the same issue - HomeKit seems to report the status of the ADT panel, but is unable to arm/disarm. In the logs I get 
> 
> 
> 
> ADT Pulse: Portal sync success.
> 
> [23/05/2022, 12:12:04] [ADT Pulse] Setting Security Panel (system-1) status from disarmed to stay...
> 
> ADT Pulse: Performing portal sync...
> 
> ADT Pulse: Setting device status...
> 
> ADT Pulse: Response path -> /myhome/23.0.0-99/Ajax/SyncCheckServ?t=1653322324055
> 
> ADT Pulse: Response path matches -> true
> 
> ADT Pulse: Portal sync success.
> 
> ADT Pulse: Response path -> /myhome/23.0.0-99/access/error.jsp
> 
> ADT Pulse: Response path matches -> false
> 
> ADT Pulse: Set device status to stay failed.
> 
> [23/05/2022, 12:12:05] [ADT Pulse] Set device status failed.

Confirmed the same for all security panel change actions

**Comment by mrjackyliang** [comment]
> @mrjackyliang any idea of a workaround outside of using the adt pulse app while a solution is found?

At the moment, no. Unless another developer comes forward and helps out, it's only me that's working on it.

Going to take a bit longer to fix because I don't currently live in a location with an ADT Pulse system.

**Comment by Mike92115** [comment]
If you need logs or someone to test, let me know I am happy to help.  Can’t help with the code but testing I am happy to do.

Thank you,
Mike

**Comment by ibogost** [comment]
I have Pulse and can work on the code too if you'll help me get my bearings.

**Comment by RodneyCapron** [comment]
It appears that one of the developers for the Home Assistant integration has figured out the issue:

https://github.com/adt-pulse-mqtt/adt-pulse-mqtt/issues/52

**Comment by mrjackyliang** [comment]
> It appears that one of the developers for the Home Assistant integration has figured out the issue:
> 
> 
> 
> https://github.com/adt-pulse-mqtt/adt-pulse-mqtt/issues/52

Oh noice. Saves me time on the fix later. I think I can release the fix by today EOD.

**Comment by kmnedd** [comment]
That would be most appreciated!

Kevin

Sent via iPhone

**Comment by mrjackyliang** [comment]
Ah so now that I understand the issue a bit more, is that the upgrade forces the link to have a `sat` code every time you arm or disarm. Back then, it only required it if you had sensors open and needed to force the action.

**Comment by mrjackyliang** [comment]
Code is fixed. Simply just added one more request on top when requesting the `sat` code and kept the old code intact.

**Comment by jastorino82** [comment]
Thank you Jacky, we all appreciate your work on this project!

**Comment by mrjackyliang** [comment]
> Is there anything we need to do to update and get it working?

Nothing at the moment, going to release the update in a bit and will let you know!

**Comment by knedd** [comment]
Thanks Jacky...looking forward to the update!

**Comment by mrjackyliang** [comment]
Version `2.1.3` is out! Thanks for the help and for reporting this issue to me :) Closing for now.

**Comment by knedd** [comment]
Again thanks!  Works like a charm!
